sdcm/utils/data_validator.py

```diff
@@ -469,6 +499,171 @@ def list_of_view_names_for_update_test(self):
+            query = f"select %s f" \
+                    f"rom %s where {' and '.join(missed_row_as_str)}"
+
+            row_data = {}
+            for table in data_for_validation.views[:-1]:
+                if 'after_update' in table:
+                    key = 'after'
+                    select_columns = columns_for_validation
+                elif 'actual_data' in table:
+                    key = 'actual'
+                    select_columns = columns_for_validation
+                elif 'expect' in table:
+                    key = 'expect'
+                    select_columns = select_pk_columns
+                else:
+                    key = 'before'
+                    select_columns = columns_for_validation
+
+                try:
+                    row_data.update({key: list(session.execute(query % (select_columns, table)))})
+                except Exception as error:  # pylint: disable=broad-except
+                    LOGGER.error("Query %s failed. Error: %s", query % table, error)
+
+            row_data.update({'source_all_columns': list(session.execute(query % (columns_for_validation,
+                                                                                 self.base_table_name)))})
+            result[i].update({'rows': row_data})
+
+            if row_data['before'] or row_data['after']:
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.WARNING,
+                    message=f"Row {diff_row} is found in the "
+                            f"{data_for_validation.views[0] if row_data['before'] else data_for_validation.views[1]} "
+                            f"view despite was not fetched. Test problem, failure in the fetcher\n"
+                            f"Row data in the source table: {row_data['source_all_columns']}\n"
+                            f"Row data in the expected table: {row_data['expect']}").publish()
+            else:
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.ERROR,
+                    error=f"Row {diff_row} is expected to be find in whether before (the row was not updated) or "
+                    f"after (the row was updated) update view, but was not found.\n"
+                    f"Possible reason of the problem: "
+                    f"the row was updated but with not expected value as it defined in profile.\n"
+                    f"Row data in the source table: {row_data['source_all_columns']}").publish()
+
+        with open(os.path.join(logdir, f"{data_for_validation.views[0]}_analyze_result_debug.json"), "w",
+                  encoding='utf8') as json_file:
+            try:
+                json.dump(result, json_file)
+            except:   # pylint: disable=bare-except
+                json_file.write(str(result))
+            LOGGER.info("analyze_result_debug json: %s", json_file.name)
+
+        # stop test
+        DataValidatorEvent.UpdatedRowsValidator(
```

Comment:
we always send this event ? I see no logic around it...

Reply:
This event is raised when test updated rows.
There are 3 more events (by type of validation):
```
class DataValidatorEvent(InformationalEvent, abstract=True):
    DataValidator: Type[SctEventProtocol]
    ImmutableRowsValidator: Type[SctEventProtocol]
    UpdatedRowsValidator: Type[SctEventProtocol]
    DeletedRowsValidator: Type[SctEventProtocol]
```

Reply:
`analyze_updated_data_and_save_in_file` function (where CITICAL event is raised) is called when found difference between actual and expected data only. So this critical event will not be raised when validation is passed.
I added one more validation for case, when all missed data was chosen as missed mistakenly